VSRAD.Package/DebugVisualizer/GroupIndexSelector.cs

```diff
@@ -78,7 +78,21 @@ private void OptionsChanged(object sender, PropertyChangedEventArgs e)
                         SetOptionsFromDispatchParams();
                     break;
                 case nameof(Options.VisualizerOptions.NDRange3D):
-                    RaisePropertyChanged(nameof(MaximumX));
+                    if (!_projectOptions.VisualizerOptions.NDRange3D)
+                    {
+                        RaisePropertyChanged(nameof(MaximumX));
```

Comment:
1. I don't remember why this line (`RaisePropertyChanged(...)`) was in the original code. It's not clear what it does and why it is needed, could you check what happens when you remove it?
2. If there's a reason for having this call before setting `X`, `Y`, `Z` with NDRange disabled and after setting `X`, `Y`, `Z` with NDRange enabled, then this should be reflected in code comments, otherwise the calling order should be identical (and this line can be moved outside of `if-else`).